packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,428 @@
+				<HStack alignment="left" spacing="3" justify="flex-start">
+					<Button
+						__next40pxDefaultSize
+						accessibleWhenDisabled
+						variant="primary"
+						onClick={ () => onSubmit( inputComment ) }
+						size="compact"
+						disabled={
+							0 === sanitizeCommentString( inputComment ).length
+						}
+					>
+						{ thread
+							? _x( 'Update', 'verb' )
+							: _x( 'Reply', 'Add reply comment' ) }
+					</Button>
+					<Button
+						__next40pxDefaultSize
+						onClick={ onCancel }
+						size="compact"
+					>
+						{ _x( 'Cancel', 'Cancel comment edit' ) }
+					</Button>
+				</HStack>
+			</VStack>
+		</>
+	);
+}
+
+/**
+ * Renders a confirmation notice component.
+ *
+ * @param {Object}   props                - The component props.
+ * @param {string}   props.confirmMessage - The confirmation message to display. Defaults to 'Are you sure?' if not provided.
+ * @param {Function} props.confirmAction  - The action to perform when the confirm button is clicked.
+ * @param {Function} props.discardAction  - The action to perform when the discard button is clicked.
+ * @return {JSX.Element} The confirmation notice component.
+ */
+function ConfirmNotice( { confirmMessage, confirmAction, discardAction } ) {
+	return (
+		<VStack
+			// translators: title for the confirmation overlay
+			title={ __( 'Confirm' ) }
+			className="editor-collab-sidebar-panel__thread-overlay"
+			spacing="0"
+			justify="space-between"
+		>
+			<p>
+				{ confirmMessage ??
+					// translators: message displayed when confirming an action
+					__( 'Are you sure?' ) }
+			</p>
+			<HStack>
+				<Button
+					__next40pxDefaultSize
+					variant="primary"
+					onClick={ confirmAction }
+					size="compact"
+				>
+					{ __( 'Yes' ) }
+				</Button>
+				<Button
+					__next40pxDefaultSize
+					onClick={ discardAction }
+					size="compact"
+				>
+					{ __( 'No' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+/**
+ * Renders the header of a comment in the collaboration sidebar.
+ *
+ * @param {Object}   props           - The component props.
+ * @param {Object}   props.thread    - The comment thread object.
+ * @param {Function} props.onResolve - The function to resolve the comment.
+ * @param {Function} props.onEdit    - The function to edit the comment.
+ * @param {Function} props.onDelete  - The function to delete the comment.
+ * @param {Function} props.onReply   - The function to reply to the comment.
+ * @param {string}   props.status    - The status of the comment.
+ * @return {JSX.Element} The rendered comment header.
+ */
+function CommentHeader( {
+	thread,
+	onResolve,
+	onEdit,
+	onDelete,
+	onReply,
+	status,
+} ) {
+	const dateSettings = getDateSettings();
+	const [ dateTimeFormat = dateSettings.formats.time ] = useEntityProp(
+		'root',
+		'site',
+		'time_format'
+	);
+
+	const memorizedMoreActions = useMemo( () => {
+		return [
+			{
+				title: _x( 'Edit', 'Edit comment' ),
+				onClick: onEdit,
```

Comment:
I see these are declared inline (see the `CommentBoard` component). Is it possible this memorization won't work because the callbacks will be re-declared on every re-render?